src/routes/trading-view/[chain]/__layout.svelte

```diff
@@ -0,0 +1,12 @@
+<script context="module">
+  import { checkChainMaintenance } from "$lib/chain/maintenance";
+
+  /** @type {import('@sveltejs/kit').Load} */
+  export function load({ params }) {
+    // Check if we should tell user to go away from this page
+    checkChainMaintenance(params.chain);
+    return {};
+  }
+</script>
+
+<slot />
```

Comment:
(my suggested change) – DRY-up the `checkChainMaintenance` calls from various page components into single `__layout.svelte`